marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getMatchingPropertyValues.sjs

```diff
@@ -68,20 +69,29 @@ if(referenceType === 'element') {
 } else if(referenceType === 'field') {
   query = cts.fieldReference(propertyPath);
 } else if(referenceType === 'collection') {
-  query = cts.collectionReference();
+  const collectionsToIgnore = Artifacts.getAllArtifactCollections();
+  collectionsToIgnore.push("http://marklogic.com/entity-services/models", "hub-core-artifact");
+
+  const allCollectionsQuery = cts.collectionQuery(cts.collections());
+  const ignoredCollectionsQuery = queryObj.ignoreArtifactCollections ? cts.collectionQuery(collectionsToIgnore) : cts.collectionQuery([]);
+  query = cts.andNotQuery(allCollectionsQuery, ignoredCollectionsQuery);
```

Comment:
Just do a cts.notQuery on a cts.collectionQuery that contains each of the collections you want to ignore. "cts.collections()" may be extremely slow when there are millions of collections.